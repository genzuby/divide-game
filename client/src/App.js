import React, { useState, useEffect } from 'react';
import { connect } from 'react-redux';
import Login from './components/Login';
import Game from './components/Game';
import { SocketProvider } from './contexts/SocketContext';
import { GameInfoProvider } from './contexts/GameInfoContext';

function App({ loginInfo }) {
  const [playerId, setPlayerId] = useState('');

  useEffect(() => {
    setPlayerId(loginInfo.userId);
  }, [loginInfo]);

  const appmain = (
    <SocketProvider id={playerId}>
      <GameInfoProvider id={playerId}>
        <Game />
      </GameInfoProvider>
    </SocketProvider>
  );

  return <div data-test="appComponent">{playerId ? appmain : <Login />}</div>;
}

const mapStateToProps = ({ loginInfo }) => {
  return { loginInfo };
};

export default connect(mapStateToProps)(App);
