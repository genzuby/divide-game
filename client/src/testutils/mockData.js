export const loginInfo = {
  isAvailable: true,
  userId: 'tomato',
  userKey: '12345678',
};

export const playerList = [
  {
    isAvailable: true,
    _id: '5fa7270bda31c407a193853a',
    userId: 'genzuby',
    userKey: '67012659-cd81-4f9e-843f-580d7b212b2b',
    __v: 0,
  },
  {
    isAvailable: true,
    _id: '5fa7274f019aaa08d961f6d8',
    userId: 'hoddie',
    userKey: '0f6b6dd6-2a5b-4718-b397-ecc7378beb3f',
    __v: 0,
  },
  {
    isAvailable: true,
    _id: '5fa7cdfe70559d085699e100',
    userId: 'hyunjoo',
    userKey: '0ae42d19-6dc5-4d9d-b2ba-52a6929eddbb',
    __v: 0,
  },
  {
    isAvailable: true,
    _id: '5fa72720da31c407a193853b',
    userId: 'shin',
    userKey: '3538278a-02a6-403a-9020-0c022cb6031a',
    __v: 0,
  },
  {
    isAvailable: true,
    _id: '5fa7dab888dcad0a93d567af',
    userId: 'test1234',
    userKey: 'c02c79e4-6cbd-4592-b069-be7be7fe9d66',
    __v: 0,
  },
  {
    isAvailable: true,
    _id: '5fa7ce0270559d085699e101',
    userId: 'tomato',
    userKey: '34d37561-562d-425d-8265-3c5d340fa43c',
    __v: 0,
  },
];

export const gamecontext_login = {
  gameInfo: {
    myid: 'tomato',
  },
};
