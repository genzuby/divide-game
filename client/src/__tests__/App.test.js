import React from 'react';
import { shallow } from 'enzyme';
import '../testutils/setup';
import { loginInfo } from '../testutils/mockData';
import { findByTestAttr, testStore } from '../testutils/utils';
import App from '../App';

const setUp = (initialState = {}) => {
  const store = testStore(initialState);
  const wrapper = shallow(<App store={store} />)
    .childAt(0)
    .dive();
  return wrapper;
};

describe('Login page render', () => {
  let wrapper;
  beforeEach(() => {
    const initialState = {};
    wrapper = setUp(initialState);
  });

  it('Should render without errors', () => {
    const component = findByTestAttr(wrapper, 'appComponent');
    expect(component.length).toBe(1);
  });

  it('renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});

describe('Game Page render', () => {
  let wrapper;
  beforeEach(() => {
    const initialState = { loginInfo };
    wrapper = setUp(initialState);
  });

  it('Should render without errors', () => {
    const component = findByTestAttr(wrapper, 'appComponent');
    expect(component.length).toBe(1);
  });

  it('renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
