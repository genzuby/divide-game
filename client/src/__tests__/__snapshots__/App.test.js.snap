// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component render renders correctly 1`] = `ShallowWrapper {}`;

exports[`Game Page render renders correctly 1`] = `ShallowWrapper {}`;

exports[`Login page render renders correctly 1`] = `ShallowWrapper {}`;
